soroban-env-common/src/val.rs

```diff
@@ -323,8 +315,6 @@ impl<E: Env> Compare<Bool> for E {
 // Declare a few extra object wrapper types that don't live anywhere else.
 declare_tag_based_object_wrapper!(VecObject);
 declare_tag_based_object_wrapper!(MapObject);
-declare_tag_based_object_wrapper!(ContractExecutableObject);
```

Comment:
Why is `ContractExecutableObject` removed as a host object type?

Reply:
> we just don't need it (so there is not object for contract instance)

Yeah this is what I wasn't sure. So the only place `ContractInstance` exist is in the storage in the xdr form, and there is no need for it as a host object?

Reply:
Yes, and I wish it wasn't even in `ScVal`, but that would require restructuring `ContractDataEntry` or introducing a new LE type.

Reply:
Actually, we could make this a pointer - that would decrease the ScVal size at cost of slightly increasing the instance entries only, WDYT @sisuresh @jayz22 ?